src/main/java/weather/CurrentWeatherController.java

```diff
@@ -0,0 +1,42 @@
+package weather;
+
+import io.reactivex.rxjava3.schedulers.Schedulers;
+import weather.current.CurrentWeather;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.swing.*;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class CurrentWeatherController {
```

Comment:
I know it wasn't part of the original work but can you write a test for this?

Reply:
Yes! I just did and found a null pointer exception in the Controller, so I tested a whole bunch of things and found I needed to update the setCurrentWeather method to have an if statement that it will only execute once main is full. Test is done and coming shortly